RecoEgamma/EgammaIsolationAlgos/src/EgammaL1TkIsolation.cc

```diff
@@ -0,0 +1,72 @@
+#include "RecoEgamma/EgammaIsolationAlgos/interface/EgammaL1TkIsolation.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+EgammaL1TkIsolation::TrkCuts::TrkCuts(const edm::ParameterSet& para) {
+  minPt = para.getParameter<double>("minPt");
+  auto sq = [](double val) { return val * val; };
+  minDR2 = sq(para.getParameter<double>("minDR"));
+  maxDR2 = sq(para.getParameter<double>("maxDR"));
+  minDEta = para.getParameter<double>("minDEta");
+  maxDZ = para.getParameter<double>("maxDZ");
+}
+
+edm::ParameterSetDescription EgammaL1TkIsolation::TrkCuts::pSetDescript() {
+  edm::ParameterSetDescription desc;
+  desc.add<double>("minPt", 2.0);
+  desc.add<double>("maxDR", 0.3);
+  desc.add<double>("minDR", 0.01);
+  desc.add<double>("minDEta", 0.003);
+  desc.add<double>("maxDZ", 0.7);
+  return desc;
+}
+
+EgammaL1TkIsolation::EgammaL1TkIsolation(const edm::ParameterSet& para)
+    : barrelCuts_(para.getParameter<edm::ParameterSet>("barrelCuts")),
+      endcapCuts_(para.getParameter<edm::ParameterSet>("endcapCuts")) {}
+
+edm::ParameterSetDescription EgammaL1TkIsolation::pSetDescript() {
+  edm::ParameterSetDescription desc;
+  desc.add("barrelCuts", TrkCuts::pSetDescript());
+  desc.add("endcapCuts", TrkCuts::pSetDescript());
+  return desc;
+}
+
+std::pair<int, double> EgammaL1TkIsolation::calIsol(const reco::TrackBase& trk, const L1TrackCollection& tracks) const {
+  return calIsol(trk.eta(), trk.phi(), trk.vz(), tracks);
+}
+
+std::pair<int, double> EgammaL1TkIsolation::calIsol(const double objEta,
+                                                    const double objPhi,
+                                                    const double objZ,
+                                                    const L1TrackCollection& tracks) const {
+  double ptSum = 0.;
+  int nrTrks = 0;
+
+  const TrkCuts& cuts = std::abs(objEta) < 1.5 ? barrelCuts_ : endcapCuts_;
```

Comment:
magic number 1.5 should be named constant (or obtained from geometry...)

Reply:
made it configurable as its more a parameter than a strict geometry based variable